### DISCUSSION TITLE
How to get WebRTC stream to play in video player

### DISCUSSION THREAD

**Discussion by dillanspencer** [bluenviron/mediamtx#2230] [Questions] [answered]
I am hoping to find the best way to display a live rtsp stream in a html video player. Right now I am using an embedded iframe to view the webrtc stream using the browser. But I want to get the stream into a video player so I have direct access to the controls of the player. Can anyone help or give me some guidance?

**Comment by aler9** [comment, marked as answer]
Hello, as written in the README, you can copy and paste the code of the WebRTC read page into a HTML file, served by an HTTP server, and edit it as you like.

**Reply by aler9** [reply]
<s>https://github.com/bluenviron/mediamtx/blob/main/internal/core/webrtc_read_index.html</s>

**Reply by morris-bah** [reply]
Broken link!

**Reply by HouCoder** [reply]
<s>It was removed somehow, you can still find it in the previous commit - https://github.com/bluenviron/mediamtx/blob/72e74b6456c147f5988705a5cb03ae94227e192c/internal/core/webrtc_read_index.html</s>

**Reply by aler9** [reply]
Nothing was removed, the right link and instructions are written in the [README](https://github.com/bluenviron/mediamtx?tab=readme-ov-file#web-browsers-1). Discussions are not valid reference since they can be out of date, while the README is not.

**Reply by thewh1teagle** [reply]
I tried to use webrtc sample, but looks like the initial connection is pretty slow.
Is there alternative for webrtc for playing rtsp in the web browser?
For instance, mpeg or some other codec which works natively?
I run the media server on the same PC which display it in the browser.

**Reply by macrokernel** [reply]
I was able to find the file at https://raw.githubusercontent.com/bluenviron/mediamtx/main/internal/servers/webrtc/read_index.html
